In [48]:
!pip install opendatasets

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [49]:
import opendatasets as od

In [50]:
train_dir='/content/train.csv'

In [51]:
!ls -lh {train_dir}

-rw-r--r-- 1 root root 8.8M Sep  4 02:00 /content/train.csv


In [52]:
!wc -l {train_dir}

20501 /content/train.csv


In [53]:
!head {train_dir}

id,type,locality,activation_date,latitude,longitude,lease_type,gym,lift,swimming_pool,negotiable,furnishing,parking,property_size,property_age,bathroom,facing,cup_board,floor,total_floor,amenities,water_supply,building_type,balconies,rent
ff8081815733a243015733b2876600a6,BHK2,"Cauvery Colony,Koramangala",21-04-2018 14:44,12.9344709,77.634471,FAMILY,1,1,1,0,SEMI_FURNISHED,BOTH,1250,25,2,E,2,6,12,"{""LIFT"":true,""GYM"":true,""INTERNET"":true,""AC"":false,""CLUB"":true,""INTERCOM"":true,""POOL"":true,""CPA"":true,""FS"":true,""SERVANT"":false,""SECURITY"":true,""SC"":true,""GP"":true,""PARK"":false,""RWH"":true,""STP"":true,""HK"":false,""PB"":true,""VP"":true}",CORP_BORE,AP,2,40000
ff8081815ee25e15015ee50004da2acd,BHK2,Bellandur,09-10-2017 12:48,12.92955671,77.6722797,ANYONE,0,1,0,1,SEMI_FURNISHED,BOTH,1400,4,2,NE,2,3,4,"{""LIFT"":true,""GYM"":false,""INTERNET"":false,""AC"":false,""CLUB"":false,""INTERCOM"":false,""POOL"":false,""CPA"":false,""FS"":false,""SERVANT"":false,""SECURITY"":

In [54]:
s='id,type,locality,activation_date,latitude,longitude,lease_type,gym,lift,swimming_pool,negotiable,furnishing,parking,property_size,property_age,bathroom,facing,cup_board,floor,total_floor,amenities,water_supply,building_type,balconies,rent'
cols=[]
cols=s.split(',')
cols

['id',
 'type',
 'locality',
 'activation_date',
 'latitude',
 'longitude',
 'lease_type',
 'gym',
 'lift',
 'swimming_pool',
 'negotiable',
 'furnishing',
 'parking',
 'property_size',
 'property_age',
 'bathroom',
 'facing',
 'cup_board',
 'floor',
 'total_floor',
 'amenities',
 'water_supply',
 'building_type',
 'balconies',
 'rent']

In [55]:
len(cols)

25

In [56]:
import pandas as pd

In [57]:
df=pd.read_csv(train_dir, usecols=cols)

In [58]:
df.head(20)

id  type  \
0   ff8081815733a243015733b2876600a6  BHK2   
1   ff8081815ee25e15015ee50004da2acd  BHK2   
2   ff80818163f29c9e0163f46770b873e7  BHK3   
3   ff80818164b68a700164b74b1b247a1d  BHK1   
4   ff80818163b1faf00163b4b9b8a163b1  BHK3   
5   ff8081815f48e4aa015f4dd1a9c6696b  BHK1   
6   ff808181647d728001647e3cd0ed458d  BHK3   
7   ff80818164e9cd1b0164ea07d09114b7  BHK2   
8   ff808181624e0deb01624e6b3426131c  BHK2   
9   ff808181620eb61601620f1b79a42734  BHK2   
10  ff8081815c116660015c119a34fd0829  BHK2   
11  ff8081815f887e60015f8b497ac62da0  BHK2   
12  ff8081815b3ce753015b3deabfae2e70  BHK2   
13  ff8081815efff8db015f0097a4a946bf  BHK2   
14  ff808181645efb5c01645f8463e227fd  BHK1   
15  ff808181620eb34b0162133b0dbe5f16  BHK2   
16  ff8081816483b0ba01648479cec8103a  BHK3   
17  ff8081815d851c3f015d8d3a9b3e5361  BHK3   
18  ff8081816058ce47016058d4e68f0308  BHK2   
19  ff80818163dffb5f0163e00d3cdd0429  BHK2   

                                             locality   activation_date  \
0                          Cauvery Colony,Koramangala  21-04-2018 14:44   
1                                           Bellandur  09-10-2017 12:48   
2                                       Thiruvanmiyur  12-06-2018 22:02   
3                                           Attiguppe  20-07-2018 16:22   
4                                           Kodihalli  31-05-2018 17:41   
5                                Seetharampalya,Hoodi  24-10-2017 18:29   
6                       Narayanappa Garden,Whitefield  12-07-2018 20:29   
7                               HSR Layout 5th Sector  30-07-2018 16:01   
8                                        Banashankari  26-03-2018 14:11   
9                                          Whitefield  10-03-2018 17:17   
10                                           RR Nagar  11-07-2017 14:58   
11                             ITPL Residential Flats  05-11-2017 16:14   
12                              Raja Rajeshwari Nagar  27-08-2017 16:55   
13  Prestige Shanthinikethan, Bengaluru, Karnataka...  09-10-2017 16:59   
14                                      Murugeshpalya  05-07-2018 15:18   
15                                     Sheshadripuram  11-03-2018 15:39   
16                                           Cox Town  15-07-2018 17:08   
17                                      BTM 2nd Stage  14-08-2017 12:02   
18                                     Chikkakannalli  15-12-2017 14:53   
19                                              agara  08-06-2018 21:28   

     latitude  longitude lease_type  gym  lift  swimming_pool  ...  bathroom  \
0   12.934471  77.634471     FAMILY    1     1              1  ...         2   
1   12.929557  77.672280     ANYONE    0     1              0  ...         2   
2   12.982870  80.262012     FAMILY    0     1              0  ...         3   
3   12.955991  77.531634     FAMILY    0     0              0  ...         1   
4   12.963903  77.649446     FAMILY    0     0              0  ...         3   
5   12.986196  77.718314     FAMILY    1     1              1  ...         2   
6   12.967144  77.750662     ANYONE    1     1              1  ...         3   
7   12.917414  77.631764     ANYONE    0     0              0  ...         2   
8   12.923301  77.556324     ANYONE    0     0              0  ...         2   
9   12.972697  77.751880     ANYONE    1     1              0  ...         2   
10  12.912039  77.510950     FAMILY    0     0              0  ...         2   
11  12.986175  77.735956     ANYONE    1     1              0  ...         2   
12  12.925034  77.517115     ANYONE    0     1              0  ...         2   
13  12.989002  77.727370     FAMILY    1     1              1  ...         2   
14  12.954775  77.650037   BACHELOR    0     0              0  ...         1   
15  12.986517  77.574781     FAMILY    0     0              0  ...         1   
16  12.998526  77.631007     FAMILY    0     0              0  ...         2   
17  12.908240  77.607409     ANYONE    0     0     

In [59]:
amen=df['amenities'][0]

In [60]:
amen

'{"LIFT":true,"GYM":true,"INTERNET":true,"AC":false,"CLUB":true,"INTERCOM":true,"POOL":true,"CPA":true,"FS":true,"SERVANT":false,"SECURITY":true,"SC":true,"GP":true,"PARK":false,"RWH":true,"STP":true,"HK":false,"PB":true,"VP":true}'

In [61]:
type(amen)

str

In [62]:

amen=amen.split(',')

In [63]:
amen

['{"LIFT":true',
 '"GYM":true',
 '"INTERNET":true',
 '"AC":false',
 '"CLUB":true',
 '"INTERCOM":true',
 '"POOL":true',
 '"CPA":true',
 '"FS":true',
 '"SERVANT":false',
 '"SECURITY":true',
 '"SC":true',
 '"GP":true',
 '"PARK":false',
 '"RWH":true',
 '"STP":true',
 '"HK":false',
 '"PB":true',
 '"VP":true}']

In [64]:
amen_fields=list()
for i in amen:
  # print(i)
  a=(i.split(':'))
  amen_fields.append(a[0])


In [65]:
amen_fields[0]='"LIFT"'

In [66]:
amen_fields

['"LIFT"',
 '"GYM"',
 '"INTERNET"',
 '"AC"',
 '"CLUB"',
 '"INTERCOM"',
 '"POOL"',
 '"CPA"',
 '"FS"',
 '"SERVANT"',
 '"SECURITY"',
 '"SC"',
 '"GP"',
 '"PARK"',
 '"RWH"',
 '"STP"',
 '"HK"',
 '"PB"',
 '"VP"']

In [67]:
amenities=list()
for i in amen_fields:
  x=i.replace('"', "", 2)
  amenities.append(x)

In [68]:
amenities

['LIFT',
 'GYM',
 'INTERNET',
 'AC',
 'CLUB',
 'INTERCOM',
 'POOL',
 'CPA',
 'FS',
 'SERVANT',
 'SECURITY',
 'SC',
 'GP',
 'PARK',
 'RWH',
 'STP',
 'HK',
 'PB',
 'VP']

In [69]:
import numpy as np


In [70]:
# print(distinct)
x = np.array(df['building_type'].tolist())
print(np.unique(x))

['AP' 'GC' 'IF' 'IH']


In [71]:
build_dict={'IH':3,'IF':2,'AP':1,'GC':0}

In [72]:
x = np.array(df['facing'].tolist())
print(np.unique(x))

['E' 'N' 'NE' 'NW' 'S' 'SE' 'SW' 'W']


In [95]:
face_dict={'N':7,'E':6,'NE':5,'W':4,'NW':3,'SE':2,'SW':1,'S':0}

In [73]:
x = np.array(df['negotiable'].tolist())
print(np.unique(x))

[0 1]


In [74]:
x = np.array(df['furnishing'].tolist())
print(np.unique(x))

['FULLY_FURNISHED' 'NOT_FURNISHED' 'SEMI_FURNISHED']


In [75]:
furnish_dict={'FULLY_FURNISHED':2,'SEMI_FURNISHED':1,'NOT_FURNISHED':0}

In [76]:
x = np.array(df['water_supply'].tolist())
print(np.unique(x))

['BOREWELL' 'CORPORATION' 'CORP_BORE']


In [77]:
water_dict={'CORP_BORE':2,'CORPORATION':1,'BOREWELL':0}

In [78]:
x = np.array(df['type'].tolist())
print(np.unique(x))

['BHK1' 'BHK2' 'BHK3' 'BHK4' 'BHK4PLUS' 'RK1']


In [32]:
type_dict={'BHK4PLUS':5,'BHK4':4,'BHK3':2,'BHK2':2,'BHK1':1,'RK1':0}

In [33]:
#parking furnishing facing


SyntaxError: ignored

In [79]:
x = np.array(df['parking'].tolist())
print(np.unique(x))

['BOTH' 'FOUR_WHEELER' 'NONE' 'TWO_WHEELER']


In [80]:
park_dict={'BOTH':3,'FOUR_WHEELER':2,'TWO_WHEELER':1,'NONE':0}

In [81]:
need_cols1=['id',
 'type',
 'gym',
 'lift',
 'swimming_pool',
 'negotiable',
 'furnishing',
 'parking',
 'property_size',
 'property_age',
 'bathroom',
 'facing',
 'cup_board',
 'floor',
 'total_floor',
 'water_supply',
 'building_type',
 'balconies',
 'rent']


In [82]:
df1=pd.read_csv(train_dir, usecols=need_cols1)

In [83]:
df1.head(10)

id  type  gym  lift  swimming_pool  \
0  ff8081815733a243015733b2876600a6  BHK2    1     1              1   
1  ff8081815ee25e15015ee50004da2acd  BHK2    0     1              0   
2  ff80818163f29c9e0163f46770b873e7  BHK3    0     1              0   
3  ff80818164b68a700164b74b1b247a1d  BHK1    0     0              0   
4  ff80818163b1faf00163b4b9b8a163b1  BHK3    0     0              0   
5  ff8081815f48e4aa015f4dd1a9c6696b  BHK1    1     1              1   
6  ff808181647d728001647e3cd0ed458d  BHK3    1     1              1   
7  ff80818164e9cd1b0164ea07d09114b7  BHK2    0     0              0   
8  ff808181624e0deb01624e6b3426131c  BHK2    0     0              0   
9  ff808181620eb61601620f1b79a42734  BHK2    1     1              0   

   negotiable       furnishing      parking  property_size  property_age  \
0           0   SEMI_FURNISHED         BOTH           1250            25   
1           1   SEMI_FURNISHED         BOTH           1400             4   
2           0   SEMI_FURNISHED         BOTH           1350             6   
3           1   SEMI_FURNISHED  TWO_WHEELER            600             3   
4           1   SEMI_FURNISHED         BOTH           1500            15   
5           1   SEMI_FURNISHED         BOTH           1080             0   
6           1  FULLY_FURNISHED         BOTH           1895             5   
7           1   SEMI_FURNISHED         BOTH           1000            10   
8           0   SEMI_FURNISHED         NONE            900            10   
9           1   SEMI_FURNISHED         BOTH           1290             4   

   bathroom facing  cup_board  floor  total_floor water_supply building_type  \
0         2      E          2      6           12    CORP_BORE            AP   
1         2     NE          2      3            4  CORPORATION            AP   
2         3      E          3      1            5    CORP_BORE            AP   
3         1      E          1      1            2  CORPORATION            IH   
4         3      E          4      0            0  CORPORATION            IH   
5         2      E          1      3            5    CORP_BORE            AP   
6         3     NE          5      3            9  CORPORATION            AP   
7         2      S          2      0            2    CORP_BORE            IH   
8         2      S          2      0            1  CORPORATION            IH   
9         2      E          4      2            4    CORP_BORE            AP   

   balconies   rent  
0          2  40000  
1          2  22000  
2          3  28000  
3          0   8000  
4          1  45000  
5          2  18000  
6          3  32000  
7          0  22000  
8          0  10000  
9          2  19500

In [84]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20500 entries, 0 to 20499
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   id             20500 non-null  object
 1   type           20500 non-null  object
 2   gym            20500 non-null  int64 
 3   lift           20500 non-null  int64 
 4   swimming_pool  20500 non-null  int64 
 5   negotiable     20500 non-null  int64 
 6   furnishing     20500 non-null  object
 7   parking        20500 non-null  object
 8   property_size  20500 non-null  int64 
 9   property_age   20500 non-null  int64 
 10  bathroom       20500 non-null  int64 
 11  facing         20500 non-null  object
 12  cup_board      20500 non-null  int64 
 13  floor          20500 non-null  int64 
 14  total_floor    20500 non-null  int64 
 15  water_supply   20500 non-null  object
 16  building_type  20500 non-null  object
 17  balconies      20500 non-null  int64 
 18  rent           20500 non-n

In [85]:
df1.describe()

gym          lift  swimming_pool    negotiable  property_size  \
count  20500.000000  20500.000000   20500.000000  20500.000000   20500.000000   
mean       0.226732      0.361463       0.174927      0.711512    1061.829463   
std        0.418728      0.480436       0.379914      0.453070     591.419631   
min        0.000000      0.000000       0.000000      0.000000       1.000000   
25%        0.000000      0.000000       0.000000      0.000000     700.000000   
50%        0.000000      0.000000       0.000000      1.000000    1047.000000   
75%        0.000000      1.000000       0.000000      1.000000    1281.000000   
max        1.000000      1.000000       1.000000      1.000000   50000.000000   

       property_age      bathroom     cup_board         floor   total_floor  \
count  20500.000000  20500.000000  20500.000000  20500.000000  20500.000000   
mean       5.770634      1.858146      4.553512      1.873561      3.801610   
std        6.350271      0.710822    335.416267      2.184927      3.185748   
min       -1.000000      1.000000      0.000000      0.000000      0.000000   
25%        1.000000      1.000000      1.000000      1.000000      2.000000   
50%        5.000000      2.000000      2.000000      1.000000      3.000000   
75%       10.000000      2.000000      3.000000      2.000000      4.000000   
max      400.000000     21.000000  48026.000000     25.000000     26.000000   

          balconies          rent  
count  20500.000000  20500.000000  
mean       1.148976  19053.602195  
std        0.993835   8433.949300  
min        0.000000   8000.000000  
25%        0.000000  12500.000000  
50%        1.000000  17500.000000  
75%        2.000000  24000.000000  
max       13.000000  50000.000000

# a few anomalies to note: 
there are a few houses with 21 bathrooms

cupboard column has value 48026

there are houses with 13 balconies



# all of these anomalies must be taken care of during model building and website building


In [86]:
df1=df1[df1["bathroom"] < 10]
df1=df1[df1["cup_board"]<20]
df1=df1[df1["balconies"]<5]

In [87]:
for ind in df1.index:
    # print(df['Name'][ind], df['Stream'][ind])
    df1['building_type'][ind]=build_dict[df1['building_type'][ind]]




/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [88]:
type_dict

{'BHK4PLUS': 5, 'BHK4': 4, 'BHK3': 2, 'BHK2': 2, 'BHK1': 1, 'RK1': 0}

In [89]:
%%time
for ind in df1.index:
    # print(df['Name'][ind], df['Stream'][ind])
    # df1['building_type'][ind]=build_dict[df1['building_type'][ind]]
    # df1['furnishing'][ind]=furnish_dict[df1['furnishing'][ind]]
    # df1['water_supply'][ind]=water_dict[df1['water_supply'][ind]]
    df1['type'][ind]=type_dict[df1['type'][ind]]
    # df1['parking'][ind]=park_dict[df1['parking'][ind]]



/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


CPU times: user 8.85 s, sys: 588 ms, total: 9.44 s
Wall time: 8.48 s


In [90]:
furnish_dict

{'FULLY_FURNISHED': 2, 'SEMI_FURNISHED': 1, 'NOT_FURNISHED': 0}

In [91]:
%%time
for ind in df1.index:
    df1['furnishing'][ind]=furnish_dict[df1['furnishing'][ind]]
    # df1['water_supply'][ind]=water_dict[df1['water_supply'][ind]]
    # df1['parking'][ind]=park_dict[df1['parking'][ind]]

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


CPU times: user 9.63 s, sys: 813 ms, total: 10.4 s
Wall time: 9.13 s


In [92]:
%%time
for ind in df1.index:
    df1['water_supply'][ind]=water_dict[df1['water_supply'][ind]]
    # df1['parking'][ind]=park_dict[df1['parking'][ind]]

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


CPU times: user 10 s, sys: 295 ms, total: 10.3 s
Wall time: 11.5 s


In [93]:
%%time
for ind in df1.index:
    df1['parking'][ind]=park_dict[df1['parking'][ind]]

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


CPU times: user 9.24 s, sys: 764 ms, total: 10 s
Wall time: 8.76 s


In [96]:
face_dict

{'N': 7, 'E': 6, 'NE': 5, 'W': 4, 'NW': 3, 'SE': 2, 'SW': 1, 'S': 0}

In [97]:
for ind in df1.index:
    df1['facing'][ind]=face_dict[df1['facing'][ind]]

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [98]:
df1.head()

id type  gym  lift  swimming_pool  \
0  ff8081815733a243015733b2876600a6    2    1     1              1   
1  ff8081815ee25e15015ee50004da2acd    2    0     1              0   
2  ff80818163f29c9e0163f46770b873e7    2    0     1              0   
3  ff80818164b68a700164b74b1b247a1d    1    0     0              0   
4  ff80818163b1faf00163b4b9b8a163b1    2    0     0              0   

   negotiable furnishing parking  property_size  property_age  bathroom  \
0           0          1       3           1250            25         2   
1           1          1       3           1400             4         2   
2           0          1       3           1350             6         3   
3           1          1       1            600             3         1   
4           1          1       3           1500            15         3   

  facing  cup_board  floor  total_floor water_supply building_type  balconies  \
0      6          2      6           12            2             1          2   
1      5          2      3            4            1             1          2   
2      6          3      1            5            2             1          3   
3      6          1      1            2            1             3          0   
4      6          4      0            0            1             3          1   

    rent  
0  40000  
1  22000  
2  28000  
3   8000  
4  45000

In [99]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 20445 entries, 0 to 20499
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   id             20445 non-null  object
 1   type           20445 non-null  object
 2   gym            20445 non-null  int64 
 3   lift           20445 non-null  int64 
 4   swimming_pool  20445 non-null  int64 
 5   negotiable     20445 non-null  int64 
 6   furnishing     20445 non-null  object
 7   parking        20445 non-null  object
 8   property_size  20445 non-null  int64 
 9   property_age   20445 non-null  int64 
 10  bathroom       20445 non-null  int64 
 11  facing         20445 non-null  object
 12  cup_board      20445 non-null  int64 
 13  floor          20445 non-null  int64 
 14  total_floor    20445 non-null  int64 
 15  water_supply   20445 non-null  object
 16  building_type  20445 non-null  object
 17  balconies      20445 non-null  int64 
 18  rent           20445 non-n

In [100]:
df1['type'] = pd.to_numeric(df1['type'])
df1['furnishing'] = pd.to_numeric(df1['furnishing'])
df1['parking'] = pd.to_numeric(df1['parking'])
df1['facing'] = pd.to_numeric(df1['facing'])
df1['water_supply'] = pd.to_numeric(df1['water_supply'])
df1['building_type'] = pd.to_numeric(df1['building_type'])






In [101]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 20445 entries, 0 to 20499
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   id             20445 non-null  object
 1   type           20445 non-null  int64 
 2   gym            20445 non-null  int64 
 3   lift           20445 non-null  int64 
 4   swimming_pool  20445 non-null  int64 
 5   negotiable     20445 non-null  int64 
 6   furnishing     20445 non-null  int64 
 7   parking        20445 non-null  int64 
 8   property_size  20445 non-null  int64 
 9   property_age   20445 non-null  int64 
 10  bathroom       20445 non-null  int64 
 11  facing         20445 non-null  int64 
 12  cup_board      20445 non-null  int64 
 13  floor          20445 non-null  int64 
 14  total_floor    20445 non-null  int64 
 15  water_supply   20445 non-null  int64 
 16  building_type  20445 non-null  int64 
 17  balconies      20445 non-null  int64 
 18  rent           20445 non-n

In [104]:
# df1.to_csv('processeddf.csv')

from google.colab import files

df1.to_csv('processeddf.csv')
files.download('processeddf.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [102]:
from sklearn.model_selection import train_test_split

In [95]:
train_df1,val_df1= train_test_split(df1,test_size=0.2,random_state=42)

In [96]:
len(train_df1)

16356

In [97]:
len(val_df1)

4089

In [98]:
train_df1=train_df1.dropna()
val_df1=val_df1.dropna()

In [99]:
print(len(train_df1))
print(len(val_df1))

16356
4089


In [101]:
train_df1.head(5)

id  type  gym  lift  swimming_pool  \
10087  ff8081815239c00501523afa9bd91094  BHK1    0     0              0   
14619  ff8081815c022dd3015c053bf6886366  BHK2    0     0              0   
5368   ff8081814fdb74dc014fde80bf640994  BHK2    0     0              0   
14832  ff80818162a582350162a8b4a5880000  BHK2    1     1              0   
12448  ff80818157bd3a640157bd5fa534033e   RK1    0     0              0   

       negotiable      furnishing       parking  property_size  property_age  \
10087           0   NOT_FURNISHED   TWO_WHEELER            600             1   
14619           1  SEMI_FURNISHED          BOTH           1000             0   
5368            1  SEMI_FURNISHED          BOTH           1055             1   
14832           0  SEMI_FURNISHED  FOUR_WHEELER           1200             9   
12448           0  SEMI_FURNISHED   TWO_WHEELER            230             3   

       bathroom facing  cup_board  floor  total_floor water_supply  \
10087         1      E          0      1            3  CORPORATION   
14619         2      E          2      0           13  CORPORATION   
5368          2      E          4      3            3     BOREWELL   
14832         2      W          2      2            4    CORP_BORE   
12448         1      E          1      0            3  CORPORATION   

      building_type  balconies   rent  
10087            IF          0   8500  
14619            AP          2  25000  
5368             AP          1  21500  
14832            AP          0  22500  
12448            IH          0   8000

In [103]:
val_df1.head(5)

id  type  gym  lift  swimming_pool  \
16330  ff8081816108a00501610915abc93a4b  BHK3    1     1              1   
7850   ff8081815aacc98e015aae15194752a5  BHK2    0     0              0   
13547  ff80818158de4fb60158dea009ce0ed2   RK1    0     0              0   
5565   ff80818160204e9b0160267e7faf7a7f  BHK2    0     0              0   
10606  ff808181638d2a350163909b97eb7088  BHK2    0     0              0   

       negotiable      furnishing      parking  property_size  property_age  \
16330           1  SEMI_FURNISHED         BOTH           1505             1   
7850            1  SEMI_FURNISHED  TWO_WHEELER            743            15   
13547           0  SEMI_FURNISHED  TWO_WHEELER            200            10   
5565            1  SEMI_FURNISHED         BOTH           1000             0   
10606           0  SEMI_FURNISHED  TWO_WHEELER            500            10   

       bathroom facing  cup_board  floor  total_floor water_supply  \
16330         2      N          3      3            5  CORPORATION   
7850          2      N          2      1            2  CORPORATION   
13547         1      W          1      0            3  CORPORATION   
5565          2      N          2      0            3    CORP_BORE   
10606         1      S          2      1            3    CORP_BORE   

      building_type  balconies   rent  
16330            AP          2  22000  
7850             IF          0  22000  
13547            IH          1   8250  
5565             AP          0  20000  
10606            AP          1  12500

In [104]:
input_cols=['id',
 'type',
 'gym',
 'lift',
 'swimming_pool',
 'negotiable',
 'furnishing',
 'parking',
 'property_size',
 'property_age',
 'bathroom',
 'facing',
 'cup_board',
 'floor',
 'total_floor',
 'water_supply',
 'building_type',
 'balconies',
 ]

In [105]:
target_cols=['rent']

In [108]:
train_inputs1=train_df1[input_cols]
train_inputs1.head(5)

id  type  gym  lift  swimming_pool  \
10087  ff8081815239c00501523afa9bd91094  BHK1    0     0              0   
14619  ff8081815c022dd3015c053bf6886366  BHK2    0     0              0   
5368   ff8081814fdb74dc014fde80bf640994  BHK2    0     0              0   
14832  ff80818162a582350162a8b4a5880000  BHK2    1     1              0   
12448  ff80818157bd3a640157bd5fa534033e   RK1    0     0              0   

       negotiable      furnishing       parking  property_size  property_age  \
10087           0   NOT_FURNISHED   TWO_WHEELER            600             1   
14619           1  SEMI_FURNISHED          BOTH           1000             0   
5368            1  SEMI_FURNISHED          BOTH           1055             1   
14832           0  SEMI_FURNISHED  FOUR_WHEELER           1200             9   
12448           0  SEMI_FURNISHED   TWO_WHEELER            230             3   

       bathroom facing  cup_board  floor  total_floor water_supply  \
10087         1      E          0      1            3  CORPORATION   
14619         2      E          2      0           13  CORPORATION   
5368          2      E          4      3            3     BOREWELL   
14832         2      W          2      2            4    CORP_BORE   
12448         1      E          1      0            3  CORPORATION   

      building_type  balconies  
10087            IF          0  
14619            AP          2  
5368             AP          1  
14832            AP          0  
12448            IH          0

In [110]:
train_targets=train_df1[target_cols]
train_targets.head(5)

rent
10087   8500
14619  25000
5368   21500
14832  22500
12448   8000

In [111]:
val_inputs1=val_df1[input_cols]
val_inputs1.head(5)

id  type  gym  lift  swimming_pool  \
16330  ff8081816108a00501610915abc93a4b  BHK3    1     1              1   
7850   ff8081815aacc98e015aae15194752a5  BHK2    0     0              0   
13547  ff80818158de4fb60158dea009ce0ed2   RK1    0     0              0   
5565   ff80818160204e9b0160267e7faf7a7f  BHK2    0     0              0   
10606  ff808181638d2a350163909b97eb7088  BHK2    0     0              0   

       negotiable      furnishing      parking  property_size  property_age  \
16330           1  SEMI_FURNISHED         BOTH           1505             1   
7850            1  SEMI_FURNISHED  TWO_WHEELER            743            15   
13547           0  SEMI_FURNISHED  TWO_WHEELER            200            10   
5565            1  SEMI_FURNISHED         BOTH           1000             0   
10606           0  SEMI_FURNISHED  TWO_WHEELER            500            10   

       bathroom facing  cup_board  floor  total_floor water_supply  \
16330         2      N          3      3            5  CORPORATION   
7850          2      N          2      1            2  CORPORATION   
13547         1      W          1      0            3  CORPORATION   
5565          2      N          2      0            3    CORP_BORE   
10606         1      S          2      1            3    CORP_BORE   

      building_type  balconies  
16330            AP          2  
7850             IF          0  
13547            IH          1  
5565             AP          0  
10606            AP          1

In [112]:
val_targets=val_df1[target_cols]
val_targets.head(5)

rent
16330  22000
7850   22000
13547   8250
5565   20000
10606  12500

In [113]:
from sklearn.dummy import DummyRegressor

In [114]:
dummy_regr = DummyRegressor(strategy="mean")

In [116]:
dummy_regr.fit(train_inputs1, train_targets)

DummyRegressor()

In [118]:
train_preds_dummy=dummy_regr.predict(val_inputs1)

In [119]:
from sklearn.metrics import mean_squared_error


In [123]:
train_rmse = mean_squared_error(val_targets, train_preds_dummy, squared=False)
train_rmse

8527.347208103169

In [124]:
### the rmse for the baseline model is 8527.347

In [125]:
from sklearn.linear_model import LinearRegression

In [126]:
linear_model=LinearRegression()


In [128]:
train_cols=[
 'type',
 'gym',
 'lift',
 'swimming_pool',
 'negotiable',
 'furnishing',
 'parking',
 'property_size',
 'property_age',
 'bathroom',
 'facing',
 'cup_board',
 'floor',
 'total_floor',
 'water_supply',
 'building_type',
 'balconies',
 ]

In [130]:
from sklearn import preprocessing

In [131]:
label_encoder = preprocessing.LabelEncoder()


In [148]:
train_inputs1['type']= label_encoder.fit_transform(train_inputs1['type'])
val_inputs1['type']= label_encoder.fit_transform(val_inputs1['type'])


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [149]:
train_inputs1['parking']= label_encoder.fit_transform(train_inputs1['parking'])
val_inputs1['parking']= label_encoder.fit_transform(val_inputs1['parking'])


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [150]:
train_inputs1['furnishing']= label_encoder.fit_transform(train_inputs1['furnishing'])
val_inputs1['furnishing']= label_encoder.fit_transform(val_inputs1['furnishing'])


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [151]:
train_inputs1['facing']= label_encoder.fit_transform(train_inputs1['facing'])
val_inputs1['facing']= label_encoder.fit_transform(val_inputs1['facing'])


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [152]:
train_inputs1['water_supply']= label_encoder.fit_transform(train_inputs1['water_supply'])
val_inputs1['water_supply']= label_encoder.fit_transform(val_inputs1['water_supply'])


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [153]:
train_inputs1['building_type']= label_encoder.fit_transform(train_inputs1['building_type'])
val_inputs1['building_type']= label_encoder.fit_transform(val_inputs1['building_type'])


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [145]:
linear_model.fit(train_inputs1[train_cols],train_targets)

LinearRegression()

In [154]:
linear_preds=linear_model.predict(val_inputs1[train_cols])

In [156]:
linear_rmse = mean_squared_error(val_targets, linear_preds, squared=False)
linear_rmse

6780.780115622501